BLUESPAWN-client/headers/util/permissions/permissions.h

```diff
@@ -0,0 +1,284 @@
+		*	bytes of memory
+		*/
+		static SecurityDescriptor CreateUserSID(DWORD dwSize);
+		/**
+		* Create a SecurityDescriptor to hold a GroupSID
+		*
+		* @param dwSize The size in bytes of the SID
+		*
+		* @return a SecurityDescriptor with the lpGroupSID value set to a pointer to dwSize
+		*	bytes of memory
+		*/
+		static SecurityDescriptor CreateGroupSID(DWORD dwSize);
+		/**
+		* Create a SecurityDescriptor to hold a DACL
+		* 
+		* @param dwSize The size in bytes of the dacl
+		* 
+		* @return a SecurityDescriptor with the dacl value set to a pointer to dwSize
+		*	bytes of memory
+		*/
+		static SecurityDescriptor CreateDACL(DWORD dwSize);
+		/**
+		* Create a SecurityDescriptor to hold a sacl
+		*
+		* @param dwSize The size in bytes of the sacl
+		*
+		* @return a SecurityDescriptor with the sacl value set to a pointer to dwSize
+		*	bytes of memory
+		*/
+		static SecurityDescriptor CreateSACL(DWORD dwSize);
+
+		/**
+		* Constructor to create a security descriptor from a PISECURITY_DESCRIPTOR
+		* 
+		* @param lpSecurity A PISECURITY_DESCRIPTOR object. All valid fields in lpSecurity 
+		*	will be copied to the corresponding field in the SecurityDescriptor object, if
+		*	such a field exists
+		*/
+		SecurityDescriptor(PISECURITY_DESCRIPTOR lpSecurity = nullptr);
+		
+		/*Getter for the lpUserSID field*/
+		PSID GetUserSID() const;
+		/*Getter for the lpGroupSID field*/
+		PSID GetGroupSID() const;
+		/*Getter for the dacl field*/
+		PACL GetDACL() const;
+		/*Getter for the sacl field*/
+		PACL GetSACL() const;
+	};
+
+	/*Enum for storing type of Owner an Owner object is*/
+	enum OwnerType {
+		NONE, USER, GROUP 
+	};
+
+	class Owner : public Loggable {
+	protected:
+		//Whether or not this owner is on the system
+		bool bExists;
+
+		//The user's SID structure
+		SecurityDescriptor sdSID;
+
+		//Owner's qualified name
+		std::wstring wName;
+
+		//Domain to which the user belongs
+		std::wstring wDomainName;
+
+		//The type of the owner
+		OwnerType otType;
+
+	public:
+		/**
+		* Constructor for an owner object based off name
+		*
+		* @param name A wstring containing the name of an object. Other fields will
+		*	be filled in if an owner of that name exists.
+		*/
+		Owner(IN const std::wstring& name);
+		/**
+		* Constructor for an owner object based off sid
+		* 
+		* @param sid A SecurityDescriptor with lpUserSID set to the sid of the owner. Other
+		*	fields will be filled in if an owner of that sid exists. 
+		*/
+		Owner(IN const SecurityDescriptor& sid);
+		/**
+		* Constructor for an owner object that sets wName, bExists, and otOwnerType, but no other fields
+		*
+		* @param name A wstring containing value to be copied to wName
+		* @param exists A boolean containing value to be copied ot bExists
+		* @param t An OwnerType containing value to be copied to otOwnerType
+		*/
+		Owner(IN const std::wstring& name, IN const bool& exists, IN const OwnerType& t);
```

Comment:
This should be protected or private (users shouldn't be able to say whether or not it exists)